### PR TITLE
refactor(shared): consolidate 15 redaction implementations into one module

### PR DESCRIPTION
**Pull request by GeorgeDong32**
### What this PR does

Before this PR:
Secret redaction was implemented 15 times across the codebase with 5 different markers (`<redacted>`, `[redacted]`, `[REDACTED]`, `***`, `Bearer <redacted>`) and 6 independently drifting key lists/regexes. Two of the text regexes were copy-pasted verbatim (OpenClaw vs DeepSeek diagnostics), the origin-strip URL helper existed twice, and the drift had already caused real misses in both directions (e.g. `DIFY_KEY` was covered by the stem list but not by the copy-pasted regexes; `openai-organization`/`openai-project` were covered only by one exact header set).

After this PR:
- One module, `src/shared/utils/redaction.ts`, importable from both main and renderer, exporting the five mechanism classes found in the wild: `redactRecord` (flat env/header maps), `redactDeep` (nested values with truncation + circular protection), `redactServerKey`, `redactUrlToOrigin` / `redactUrlParams`, `redactSecretText` / `redactLiteral`.
- Key knowledge lives in exactly one place: `SENSITIVE_KEY_STEMS` (substring, case-insensitive) plus `SENSITIVE_KEY_EXACT` for names no stem covers (`openai-organization`, `openai-project`).
- Scope-limited names like OAuth's `code` are passed as call-site extras (`redactUrlParams(url, ['code'])`, `redactSecretText(text, ['code'])`) so they never redact JSON body keys globally.
- All 15 local implementations deleted, including `src/main/utils/envRedaction.ts`, `src/main/utils/redactUrl.ts`, and the renderer's `cliConfig/redact.ts`; their contract tests move to `redaction.test.ts` (32 tests, including the smol-toml real-error edge cases).

**Stacked on #18650** — merge that first; this PR will retarget to `main` automatically.

Fixes #(n/a — follow-up consolidation to #18648 / #18650)

### Why we need it and why it was done in this way

The following tradeoffs were made:
- Marker unification to `<redacted>` changes user-visible text in the network devtools panel and NDJSON trace files (previously `[redacted]` / `***`). OpenRouter's protocol-level `[REDACTED]` placeholder (`openrouterReasoning`) is NOT our redaction and is untouched.
- `MarkdownResultStore`'s persisted `downloadUrl` now redacts to origin only (pathname dropped) — the jobId in the same context already identifies the resource.
- httpTrace userinfo credentials are now replaced with the marker instead of being removed.
- The devtools body viewer keeps its own traversal (full fidelity, no truncation) and only reuses `isSensitiveKey` — sharing the drifting key knowledge, not the log-oriented traversal policy.
- The free-text regex keeps a curated key alternation instead of the bare stems, because bare `KEY`/`PASS` stems over-match English words (`MONKEY = enabled`, `PASSED = true`); the TOML triple-quote and broken-quote edge cases from the renderer implementation are preserved verbatim.

The following alternatives were considered:
- Keeping per-module redaction: rejected — the DIFY_KEY / openai-* drift shows each independent list is a separate leak waiting to happen.
- Moving everything into a global logger-layer redaction: broader blast radius, better discussed separately upstream.

### Breaking changes

None for functionality. Log/trace/panel text changes as listed above.

### Special notes for your reviewer

- Two commits: `8a7d3b17` adds the module + tests; `80dd5789` migrates all 20 call-site files and deletes the old helpers.
- Behavior deltas are enumerated in the second commit message and above; every one has either an updated test expectation or a shared-test case pinning it (`code` scope, `openai-*` exact, TOML edge cases).
- Full suite green locally: 11855 main + 8967 renderer + 2996 shared/ui tests; typecheck, i18n, format, scoped lint clean.

### Checklist

- [ ] Branch: This PR targets `fix/mcp-serverkey-log-leak` (stacked on #18650); it will retarget to `main` when that merges
- [x] PR: The PR description is expressive enough and will help future contributors
- [x] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#codeforhumans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [x] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [x] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Documentation: A [user-guide update](https://docs.cherry-ai.com) is considered but not required — log/trace text only, no user-facing behavior or usage change
- [x] Self-review: I have reviewed my own code (e.g. via [`/gh-pr-review`](/.claude/skills/gh-pr-review/SKILL.md), `gh pr diff`, or GitHub UI) before requesting review from others

### Release note

```release-note
NONE
```

---

### Post-review follow-ups

- `106fbaf6` (review follow-up): all 8 stems are now pinned by a single enumeration test — the stem list is the project's only key knowledge, so removing one must fail CI; `redactSecretText`'s `extraKeys` regex injection now escapes metacharacters so a future caller cannot silently break the alternation.
- Implementation note vs the original plan: `CherryInOAuthService` keeps its local recursion (string values → `redactSecretText(value, ['code'])`, keys → `isSensitiveKey`) instead of migrating to `redactDeep` — `redactDeep` cannot carry the `['code']` extra into nested strings and does not run text patterns on string values, so a direct migration would regress `code=`/`token=` redaction inside nested strings.

---

### Rebase note

Rebased onto the parent branch's fail-closed fix (review 4943381492 on #18650): the shared `redactServerKey` now redacts **all** env/headers values instead of delegating to `redactRecord`'s key-name matching, `McpCatalogService` imports it from the shared module, and the fail-closed contract is pinned in `redaction.test.ts` (`DATABASE_URL` + custom-header regression).

---

### Comment-cap follow-up (claude-md-compliance review)

- The three over-cap inline comment blocks in `redaction.ts` (10/3/3 lines) are compressed to ≤2 lines each; the regex ordering and TOML edge-case rationale now lives next to the tests that pin it in `redaction.test.ts`.
- Added the explicit `extraKeys` regex-escaping test (a parenthesized extra key would corrupt the alternation if unescaped).

### CODE REVIEW

File: src/shared/utils/redaction.ts [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+  }
+}
+
+// Curated alternation (not the bare stems — those over-match English words); ordering
+// and value-alternative rationale lives with the edge-case tests in redaction.test.ts.
+const SECRET_KEY_VALUE_PATTERN =
+  /(["']?(?:api[_-]?key|apikey|token|secret|password|passphrase|auth|credential|cookie|session)\w*["']?\s*[:=]\s*)("""[\s\S]*?"""|'''[\s\S]*?'''|[^\r\n]*)/gi
+
+// Bearer/Basic must run before the key=value pass — see the tests for why.
+const BEARER_SCHEME_PATTERN = /\b(Bearer|Basic)\s+[^\s"',;}\]]+/gi
```
Comment (404-Page-Found) [inline src/shared/utils/redaction.ts]: **[A4]** `BEARER_SCHEME_PATTERN` (`/\b(Bearer|Basic)\s+[^\s"',;}\]]+/gi`) cannot match a Bearer/Basic token wrapped in literal quotes — e.g. `Bearer "sk-ant-quoted-secret"` passes through both the Bearer pass and the key=value pass unredacted (runtime-verified). Pre-PR, `cliConfig/redact.ts` and `CherryInOAuthService` used `Bearer\s+\S+`, which redacted quoted tokens — so the consolidation regresses exactly those two call sites (OAuth error bodies / parser messages echoing `Bearer "..."`), and this delta is missing from the enumerated behavior list. Suggested fix — add a quoted-value alternative and pin it with a test:

```js
const BEARER_SCHEME_PATTERN = /\b(Bearer|Basic)\s+("(?:\\.|[^"\\])*"|'(?:\\.|[^'\\])*'|[^\s"',;}\]]+)/gi
```

Test: `expect(redactSecretText('detail: Bearer "sk-ant-quoted-secret"')).not.toContain('sk-ant-quoted-secret')`

**Review by 404-Page-Found — CHANGES_REQUESTED** [review]
Consolidation is excellent — faithful migrations, contract tests re-homed, fail-closed redactServerKey. One requested change: the shared Bearer/Basic pattern leaks tokens wrapped in literal quotes (e.g. Bearer "sk-..."), a regression for the two call sites that previously used Bearer\s+\S+ (cliConfig TOML errors and CherryInOAuth diagnostics). See inline comment for the fix and pinning test.

**Review by 404-Page-Found — APPROVED** [review]
LGTM — 审查通过。

**审查结论**:本 PR 将散落在 main/renderer/shared 的 15 处脱敏实现合并为 src/shared/utils/redaction.ts 单一模块,迁移了全部 20 处调用点并删除了旧辅助文件。经逐调用点核对:

- 行为不变处均为逐字搬运(
edactDeep ↔ 
edactSensitive、
edactServerKey、
edactUrlToOrigin);
- 所有行为差异(标记统一为 <redacted>、userinfo 改为打标、MarkdownResultStore 仅保留 origin、devtools 保留全保真遍历)均在 PR 描述中枚举,并由更新后的测试断言锁定;
- 被删除的契约测试(TOML 三引号/坏引号边界、smol-toml 真实报错、env 脱敏、URL 脱敏)全部在新 
edaction.test.ts(30+ 用例)中保留;
- code 作用域限制与 xtraKeys 正则转义均有测试覆盖;openai-organization/openai-project 精确名修复了 httpTrace 的漂移缺口;
- 上一轮评审关于带引号 Bearer/Basic token 泄漏的评论已由最新提交 b83331b7 修复并加回归测试;
- 所有差异方向均为更严格脱敏(安全方向),未发现新增泄漏;共享层放置规则合规;CI 全部通过。

发现的问题均为非阻塞级别(devtools body 视图的过度脱敏、一处 JSDoc 与实现不符、code 额外键子串匹配、共享层导出可变数组),不影响合并。

**Review by DeJeune — APPROVED** [review]

**Review by eeee0717 — APPROVED** [review]
LGTM
